components/consentRequestForm/index.jsx

```diff
@@ -286,10 +289,7 @@ export default function ConsentRequestForm({ agentDetails, agentWebId }) {
           {`${agentName || agentWebId} will have access until`}
         </span>
         <DateInput
-          // selectedDate={selectedDate}
-          // setting date to forever for now until we add functionality to change this
-          selectedDate={null}
-          setSelectedDate={setSelectedDate}
+          selectedDate={selectedDate}
           datepickerOpen={datepickerOpen}
           setDatepickerOpen={setDatepickerOpen}
```

Comment:
this is really internal state being leaked to the parent component, I think, as the date picker automatically closes when you click outside of it; you could also have the `handleDateChange` prop actually be executed in the DateInput like:

```
const onDateChange = useCallback((newDate) => {
  setDatepickerOpen(false)
  handleDateChange(/* format(newDate as iso8601) */);
}, [setDatepickerOpen, handleDateChange]);

const onSelectForever = useCallback(() => {
    setDatepickerOpen(false)
    handleDateChange('Forever');
}, [setDatepickerOpen, handleDateChange]);
```

That way here all you have is the `handleDateChange` method which receives either `'Forever'` or an ISO 8601 date string.